src/ik_memetic.cpp

```diff
@@ -179,10 +173,7 @@ void MemeticIk::reproduce(Robot const& robot, CostFn const& cost_fn) {
 
         } else {
             // If the mating pool is empty, roll a new population member randomly.
-            for (size_t j_idx = 0; j_idx < robot.variables.size(); ++j_idx) {
-                auto const& var = robot.variables[j_idx];
-                population_[i].genes[j_idx] = rsl::uniform_real(var.clip_min, var.clip_max);
-            }
+            population_[i].genes = robot.get_random_valid_configuration();
```

Comment:
@tylerjw C++ question: Does switching this implementation from modifying elements of a pre-existing vector to returning a brand new one and overwriting the variable affect performance significantly?

Wondering if it makes more sense for this new function to accept a mutable vector by reference instead.

Reply:
This is one of the reasons that C++ std algorithms take iterators to existing containers. I don't care either way. I think this is easier to read, and I'd always bias towards easier-to-read code unless you have a benchmark showing that the performance difference matters for your use-case.

Reply:
Yeah, I think this is fine especially because it only happens when initializing a new population member and not all the time.